examples/platform/esp32/route_hook/esp_route_table.c

```diff
@@ -85,7 +85,10 @@ esp_route_entry_t * esp_route_table_add_route_entry(const esp_route_entry_t * ro
     }
     entry->preference       = route_entry->preference;
     entry->lifetime_seconds = route_entry->lifetime_seconds;
-    sys_timeout(entry->lifetime_seconds * 1000, route_timeout_handler, entry);
+    if (entry->lifetime_seconds != UINT32_MAX)
+    {
+        sys_timeout(entry->lifetime_seconds * 1000, route_timeout_handler, entry);
+    }
```

Comment:
How was the old code causing a crash?

Reply:
lwIP checks the timeout and aborts on unreasonable values. Passing in `UINT32_MAX * 1000` overflows and will trigger the abort.

Reply:
Two things then:

1. What is the interface for sys_timeout?  Don't we need to actively avoid the overflow for all values?  Is the first argument uint32?  Then this doesn't necessarily avoid overflow.
2. If lwIP has its own notion of unreasonable, don't we need to make sure that we don't exceed that bound for all inputs?

Reply:
Updated with the lwIP limit.